src/js/utils/hooks.js

```diff
@@ -42,3 +42,41 @@ export const useDidUpdateEffect = (onUpdate, deps) => {
         firstRef.current = true;
     }, deps);
 };
+
+export function getUrlSearchParams(key) {
+    const params = new URLSearchParams(window.location.search);
+    return params.get(key) || "";
+}
+
+export function updateUrlSearchParams(newTerm, key) {
+    const search = new URLSearchParams(window.location.search);
+    search.set(key, newTerm);
+    const newUrl = `${window.location.pathname}?${search.toString()}`;
+    window.history.replaceState({}, "", newUrl);
+}
+
+/**
+ * Hook for managing and synchronizing URL search parameters with a component's state
+ *
+ * @param key - The search parameter key to be managed
+ * @param defaultValue - The default value to use when the search parameter is not present in the URL
+ * @returns Object - An object containing the current value and a function to set the URL search parameter
+ */
+export function useUrlSearchParams({ key, defaultValue }) {
+    const searchTerm = getUrlSearchParams(key);
```

Comment:
This is a generic hook for managing any search parameter. Why are you using the word 'term'?